lib/util/pdf.js

```diff
@@ -28,6 +28,21 @@ exports.parse = async function parse (docOptions, callbacks) {
     map: new Map(),
   }
 
+  let pageIndexNumMap = {}
+  let firstPage
+  for (let j = 1; j <= pdfDocument.numPages; j++) {
```

Comment:
so if we are limiting the search to the first ten pages, why iterate over all of them?

Reply:
Because we don't know when the first ten pages _with page number_  will appear. There could be cover pages, table of content, foreword, preface etc before the actual content that does not have page numbers.

Reply:
Right, but the for-loop would still blindly iterate over the rest of the pages even after the dictionary has been populated. Could you figure a way to terminate the loop earlier? Eg, you could add a clause such that the for loop continues if `j <= pdfDocument.numPages && <something else>`

Reply:
```
  for (let j = 1; j <= pdfDocument.numPages; j++) {
    const page = await pdfDocument.getPage(j)
    const textContent = await page.getTextContent()

    if (Object.keys(pageIndexNumMap).length < 10) {
      pageIndexNumMap = findPageNumbers(pageIndexNumMap, page.pageIndex, textContent.items)
    } else {
      firstPage = findFirstPage(pageIndexNumMap)
      break
    }
  }
```
The loop is break in the if condition

Reply:
Not happy with the `break` since can be missed (as I have demonstrated), but okay.